pkg/agent/plugin/keymanager/v0.go

```diff
@@ -0,0 +1,75 @@
+package keymanager
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/x509"
+
+	"github.com/spiffe/spire/pkg/common/plugin"
+	"github.com/spiffe/spire/proto/spire/agent/keymanager"
+	keymanagerv0 "github.com/spiffe/spire/proto/spire/agent/keymanager/v0"
+	"google.golang.org/grpc/codes"
+)
+
+type V0 struct {
+	plugin.Facade
+
+	Plugin keymanagerv0.KeyManager
+}
+
+func (v0 V0) GenerateKey(ctx context.Context) (crypto.Signer, error) {
+	resp, err := v0.Plugin.GenerateKeyPair(ctx, &keymanagerv0.GenerateKeyPairRequest{})
+	if err != nil {
+		return nil, v0.WrapErr(err)
+	}
+
+	if resp.PrivateKey == nil {
+		return nil, v0.Error(codes.Internal, "plugin generate key pair response missing private key")
```

Comment:
plugin _generated_?

Reply:
sorry, this is poorly worded, "generate key pair" was supposed to be the noun here. I'll reword.